tests/lib/pkgdb.sh

```diff
@@ -438,6 +438,13 @@ distro_install_build_snapd(){
         if [[ "$SPREAD_SYSTEM" = ubuntu-14.04-* ]] && [ "$SPREAD_REBOOT" = 0 ]; then
             REBOOT
         fi
+    elif [ -n "$PPA_VALIDATION_NAME" ]; then
+        apt install -y snapd
+        add-apt-repository -y "$PPA_VALIDATION_NAME"
+        apt update
+        apt install -y --only-upgrade snapd
```

Comment:
If the PPA does not have an updated snapd this call will not error and we only test the distro snapd:
```
$ sudo apt install -y --only-upgrade snapd ; echo $?
Reading package lists... Done
Building dependency tree       
Reading state information... Done
snapd is already the newest version (2.37.4+18.10.1).
0 upgraded, 0 newly installed, 0 to remove and 221 not upgraded.
0
```
Adding something like:
```
apt show snapd | grep "APT-Sources: http.*ppa.launchpad.net"
```
after the install will double check that it really comes from the PPA.

Reply:
done